RHDEVDOCS-4781 - Loki (#59285)
openshift/openshift-docs: modules/logging-loki-gui-install.adoc

// Module is included in the following assemblies:
// logging/logging-loki-ocp.adoc
//
:_content-type: PROCEDURE
[id="logging-loki-gui-install_{context}"]
= Installing logging Operators using the {product-title} web console

To install and configure logging on your {product-title} cluster, additional Operators must be installed. This can be done from the Operator Hub within the web console.

{Product-title} Operators use custom resources (CR) to manage applications and their components. High-level configuration and settings are provided by the user within a CR. The Operator translates high-level directives into low-level actions, based on best practices embedded within the Operator’s logic. A custom resource definition (CRD) defines a CR and lists all the configurations available to users of the Operator. Installing an Operator creates the CRDs, which are then used to generate CRs.

.Prerequisites

> * Supported Log Store (AWS S3, Google Cloud Storage, Azure, Swift, Minio, OpenShift Data Foundation)

Author: I can't think of any instance of us referring to cost, so I'm going to go with no, but I'll change it to object store.

.. Type `OpenShift Logging` in the filter by keyword box. Choose *Red Hat OpenShift Logging* from the list of available Operators and click *Install*.

.. On the Install Operator page, under *Update channel* select *stable*.
+
--
include::snippets/logging-stable-updates-snip.adoc[]
--

As the `Red Hat OpenShift Logging` Operator is only deployable to the `openshift-logging` namespace, *Installation mode* and *Installed Namespace* is already selected. If this namespace does not already exist, it is created for you.

.. If you are creating the `openshift-logging` namespace, select the option to *Enable Operator recommended cluster monitoring on this Namespace*.
+
[NOTE]
====
If the `openshift-logging` namespace already exists, you must add the namespace label, `openshift.io/cluster-monitoring: "true"`, to enable metrics service discovery.
====

.. Under *Update approval* select *Automatic*.
+
--
include::snippets/logging-approval-strategy-snip.adoc[]
--

.. For *Console plugin* select *Enable*, then click *Install*.

. Install the `Loki Operator`:

.. In the {product-title} web console, click *Operators* -> *OperatorHub*.

> .. Type `Loki Operator` in the filter by keyword box. Choose *Loki Operator* from the list of available Operators and click *Install*.

Author: I don't think so, I think the catalog does that well enough.

> .. Type `Loki Operator` in the filter by keyword box. Choose *Loki Operator* from the list of available Operators and click *Install*.

Reviewer: Searching for "Loki Operator" results in two items in the OperatorHub: ![image](https://github.com/openshift/openshift-docs/assets/125773/ac686fbc-6363-4172-b8a8-9fa7735506a1)

The operator logos also seem to randomly switch between the two variants... I have seen both the "L" logo and our red one on both variants already. :thinking:

> .. Type `Loki Operator` in the filter by keyword box. Choose *Loki Operator* from the list of available Operators and click *Install*.

Author: Ok then yes, should definitely be mentioned, good call. Probably should file a bug, too.

.. On the Install Operator page, for *Update Channel* select *stable*.
+
--
include::snippets/logging-stable-updates-snip.adoc[]
--

As the Loki Operator must be deployed to the global operator group namespace `openshift-operators-redhat`, *Installation mode* and *Installed Namespace* will already be selected. If this namespace does not already exist, it will be created for you.

> .. Select Enable operator recommended cluster monitoring on this namespace.

Reviewer 2: ```suggestion
.. Select *Enable operator-recommended cluster monitoring on this namespace.*
```

This option sets the openshift.io/cluster-monitoring: "true" label in the Namespace object. You must select this option to ensure that cluster monitoring scrapes the openshift-operators-redhat namespace.

.. For *Update approval* select *Automatic*, then click *Install*.
+
--
include::snippets/logging-approval-strategy-snip.adoc[]
--

The Operators should now be available to all users and projects that use this cluster.

. Verify the operator installations:
.. Navigate to *Operators* -> *Installed Operators*.
.. Make sure the `openshift-logging` project is selected.

> .. In the *Status* column you should see green checks with *InstallSucceeded* the text `Up to date` below.

Reviewer 2: ```suggestion
.. In the *Status* column, verify that you see green checks with *InstallSucceeded* and the text `Up to date`, below.
```

+
[NOTE]
====

> An Operator might display a `Failed` status before the installation finishes. If the Operator install completes with an `InstallSucceeded` message, refresh the page.

Reviewer 2: ```suggestion
An Operator might display a `Failed` status before the installation finishes. If the Operator installation completes with an `InstallSucceeded` message, refresh the page.
```
